### PR TITLE
Milestone: Create our own Enumerable :rocket: 

### PR DESCRIPTION
**Pull request by ahmedeid6842**
## Description :sparkles:

This pull request adds the implementation of the `MyList` class and the `MyEnumerable` module, which mimic the functionality of the corresponding methods in Ruby's `Enumerable` module. It also includes the necessary test cases to verify the correctness of the implementation.

## Done :heavy_check_mark:

* [x] Created the `enum.rb` file, which defines the `MyEnumerable` module. :file_folder:
* [x] Implemented the `all?`, `any?`, and `filter` methods within the `MyEnumerable` module, providing functionality similar to the corresponding methods in Ruby's `Enumerable`. :rocket:
* [x] Created the `my_list.rb` file, which defines the `MyList` class. :notebook:
* [x] Implemented the `each` method within the `MyList` class, allowing it to yield successive members of the `@list` instance variable. :link:
* [x] Included the `MyEnumerable` module in the `MyList` class, enabling the usage of the `all?`, `any?`, and `filter` methods within `MyList` instances. :heavy_check_mark:
* [x] Created the `main.rb` file, which demonstrates the usage of the implemented methods. :computer:
* [x] Added test cases to verify the correctness of the implementation, as described in the project requirements. :white_check_mark:

## Testing :mag:

The following tests were performed to validate the implementation:

1. Created a `MyList` instance with the values \[1, 2, 3, 4\]. :tada:
1. Tested the `all?` method with a block that checks if each element is less than 5. The expected result is `true`. :thumbsup:
1. Tested the `all?` method with a block that checks if each element is greater than 5. The expected result is `false`. :thumbsdown:
1. Tested the `any?` method with a block that checks if any element is equal to 2. The expected result is `true`. :smiley:
1. Tested the `any?` method with a block that checks if any element is equal to 5. The expected result is `false`. :frowning:
1. Tested the `filter` method with a block that filters even elements. The expected result is a new array containing \[2, 4\]. :tada:

## Highlights :star:

* Included test cases to verify the correctness of the implementation.
* Ensured each class and module has a separate `.rb` file.
* Ran the tests and confirmed they passed successfully.
* Verified the implementation against the provided project requirements.
* Documented the changes made and provided a clear description in this pull request.

### CODE REVIEW

File: enum.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+class MyList
+  include MyEnumerable
+  def initialize(*list)
+    @list = list
+  end
+
+  def each(&block)
+    @list.each(&block)
+  end
+end
```
Comment (mikethreels) [inline enum.rb:36]: - [ ] According to the requirements Each class and module should have a separate .rb file.

File: README.md [hunk 2]
```diff
@@ -1 +1,146 @@
...
+
+I would like to thank Microverse and Ruby for making this possible.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LICENSE -->
+
+## License <a name="license"></a>
+
+This project is [MIT](./LICENSE) licensed.
```
Comment (mikethreels) [inline README.md:144]: - [ ] the link to the MIT license is not working as expected, please make sure it works as intended in the microverse template.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- good job setting up the methods.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by cynthiainga — APPROVED** [review]
## STATUS APPROVED ✔️ 🎉 ✔️ 

Hi @ahmedeid6842 & @Gh-Reza, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- [ ] Descriptive PR ✔️ 
- [ ] No linters errors ✔️ 
- [ ] All requested changes are made ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**